### PR TITLE
Count inversion in Java

### PR DESCRIPTION
**Pull request by pranjalshinde**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

As assigned by @iamrajiv for issue #1778, I have added the solution for the Counting Inversion problem using modified merge sort algorithm

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#1778

### CODE REVIEW

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@pranjalshinde Add sample input-output, space and time complexity.

File: Java/README.md [hunk 1]
```diff
@@ -151,7 +151,7 @@ _add list here_
 - [Bilinear Search](search/Bilinear_Search.java)
 
 ## Sorting
-
+- [Count INversion](sort/CountInversion.java)
```
Comment (iamrajiv) [inline Java/README.md:154]: @pranjalshinde Update this to `- [Count Inversion](sort/CountInversion.java)`.

File: Java/sort/CountInversion.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        Scanner sc = new Scanner(System.in);
+        int len = sc.nextInt();
+        int[] arr = new int[len];
+        for(int i = 0; i < len; i++){
+            arr[i] = sc.nextInt();
+        }
+        System.out.println(mergeSort(arr,0,len-1));
+    }
+    
+
```
Comment (srimanikantapalakollu) [inline Java/sort/CountInversion.java:44]: @pranjalshinde please remove the extra spaces and also add the comments that could understand by everyone.

File: Java/sort/CountInversion.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import java.util.*;
+
+public class CountInversion{
+    private static int mergeSort(int[] arr, int low, int high){
+        int count = 0;
+        if(low < high){
+            int mid = low + (high-low)/2;
+            count += mergeSort(arr, low, mid-1);
+            count += mergeSort(arr, mid+1, high);
+
```
Comment (srimanikantapalakollu) [inline Java/sort/CountInversion.java:10]: Remove extra spaces like this.! Add the necessary comments to your code.

File: Java/sort/CountInversion.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    }
+    private static int merge(int[] arr, int low, int high, int mid){
+        int cnt = 0;
+        int i = low;
+        int j = mid+1;
+
+        while(i<=mid && j <=high){
+            if(arr[i] <= arr[j]){
+                i++;
+
```
Comment (srimanikantapalakollu) [inline Java/sort/CountInversion.java:24]: Remove extra line.

File: Java/sort/CountInversion.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+            if(arr[i] <= arr[j]){
+                i++;
+                
+            }
+            else{
+                j++;
+                cnt += high-i;  
+            }
+        }
+
```
Comment (srimanikantapalakollu) [inline Java/sort/CountInversion.java:31]: Remove the extra line.

**Review by srimanikantapalakollu — CHANGES_REQUESTED** [review]

**Comment by HarshCasper** [comment]
Closing the PR because of no updates.

**Pull request closed without merging**